packages/core/src/util.ts

```diff
@@ -51,14 +51,63 @@ export function makeBacktrace(stack: string, shift = 0): BacktraceFrame[] {
           column: line.column
         }
       })
-    backtrace.splice(0, shift)
+    if (filterHbSourceCode) {
+      backtrace.splice(0, calculateBacktraceShift(backtrace))
+    }
+
     return backtrace
   } catch (_err) {
     // TODO: log error
     return []
   }
 }
 
+const DEFAULT_BACKTRACE_SHIFT = 3
+
+function isFrameFromHbSourceCode(frame: BacktraceFrame) {
+  return frame.file.indexOf('@honeybadger-io') > -1 || frame.method.indexOf('@honeybadger-io') > -1
+}
+
+/**
+ * If {@link generateStackTrace} is used, we want to exclude frames that come from
+ * Honeybadger's source code.
+ *
+ * Logic:
+ * - For each frame, increment the shift if source code is from Honeybadger
+ * - If a frame from an <anonymous> file is encountered increment the shift ONLY if between Honeybadger source code
+ *   (i.e. previous and next frames are from Honeybadger)
+ * - Exit when frame encountered is not from Honeybadger source code
+ *
+ * Note: this will not always work, especially in browser versions where code
+ *       is minified, uglified and bundled.
+ *       For those cases we default to 3:
+ *       - generateStackTrace
+ *       - makeNotice
+ *       - notify
+ */
+export function calculateBacktraceShift(backtrace: BacktraceFrame[]) {
```

Comment:
I'm not excited with this, but I think it's better than the default approach (i.e. always remove first 3 frames)